LanguageFeatures/Augmentation-libraries/defining_augmentation_A02_t02.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Augmentations do not share an import scope with the main library
+/// or each other. The libraries one augmentation imports are visible only to
+/// that file.
+///
+/// @description Checks that imports in an augmentation library are not visible
+/// in a main library even if an augmentation library exports it
```

Comment:
Do we have a test where an augmentation library 'aug_lib.dart' exports 'lib.dart' such that the augmented library 'auged_lib.dart' must implicitly export 'lib.dart', and then some other library 'main.dart' imports `auged_lib.dart' and must now have access to 'lib.dart' declarations?

The point is that we'd like to confirm that 'lib.dart' is _actually_ exported from 'auged_lib.dart', and not just ignored entirely.

Reply:
Added `defining_augmentation_A02_t06*` test